db/db_impl/db_impl_open.cc

```diff
@@ -356,6 +356,8 @@ Status DBImpl::Recover(
 
   bool is_new_db = false;
   assert(db_lock_ == nullptr);
+  std::vector<std::string> dbname_children;
+  Status dbname_children_s = env_->GetChildren(dbname_, &dbname_children);
```

Comment:
If `best_efforts_recovery == false`, we may not have to call `GetChildren()` on `dbname`.

Reply:
I think we should not call `GetChildren()` here if `best_efforts_recovery == false`.

Reply:
> I think we should not call `GetChildren()` here if `best_efforts_recovery == false`.

Note that at the end of `Recover()`, the result of `GetChildren()` will always be required if `read_only == true`, regardless of the value of `best_efforts_recovery`.

I think the only case where `Recover()` need not be called is `read_only == false && best_efforts_recovery == false`. We can still avoid calling it in this case, though.

Reply:
Yeah, `read_only == false && best_efforts_recovery == false` is the common case.